src/Psalm/Internal/Type/ArrayType.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types=1);
+
+namespace Psalm\Internal\Type;
+
+use Psalm\Type;
+
+/**
+ * @internal
+ */
+class ArrayType
+{
+    /**
+     * @return array{key: Type\Union, value: Type\Union, is_list: bool}|null
```

Comment:
This is my only quibble to an otherwise excellent PR - a few people have pointed out in the past that arrays use more memory than corresponding objects.

What if this instead returned an instance of `ArrayType` with the properties `$key`, `$value` and `$is_list`?

Reply:
Sure, that would be better